README.rst

```diff
@@ -13,37 +13,58 @@ This benchmark is dedicated to solver of the ResNet classification fitting probl
 where :math:`i` is the sample index, :math:`x_i` is the input image, :math:`y_i` is the sample label, and :math:`L` is the cross-entropy loss function.
 
 
-Install
+Use
 --------
 
 This benchmark can be run using the following commands:
 
 .. code-block::
 
    $ pip install -U benchopt
-   $ git clone https://github.com/zaccharieramzi/benchmark_resnet_classif
+   $ git clone https://github.com/benchopt/benchmark_resnet_classif
    $ benchopt run benchmark_resnet_classif
 
-Apart from the problem, options can be passed to `benchopt run`, to restrict the benchmarks to some solvers or datasets, e.g.:
+While this command would run the entire benchmark, which includes several models, datasets and solvers, sequentially,
+you can restrict the run to a specific model, dataset and/or solver by passing the corresponding arguments.
+For example if I want to run the benchmark for the ResNet18 model on CIFAR10 dataset with the Adam solver, without a validation set, and for a single random seed:
 
 .. code-block::
 
-	$ benchopt run benchmark_resnet_classif -s solver1 -d dataset2 --max-runs 10 --n-repetitions 10
+	$ benchopt run benchmark_resnet_classif -o "*18" -d "cifar[*random_state=42*with_validation=False]" -s "adam-torch[batch_size=128,coupled_weight_decay=0.0,data_aug=True,decoupled_weight_decay=0.02,*,lr_schedule=cosine]"  --max-runs 200 --n-repetitions 1
 
-Current workaround for https://github.com/benchopt/benchopt/issues/306:
+Use `benchopt run -h` for more details about these options, or visit https://benchopt.github.io/api.html.
 
-..code-block::
+Extension
+---------
+
+If you want to add a new solver, you will need probably need to inherit one of the base solver classes from PyTorch, TensorFlow or PyTorch-Lightning.
+For example, to implement a new PyTorch-based solver, you will need at the beginning of your solver class to add the following:
+
+::
+
+
+   from benchopt import safe_import_context
+
+
+   with safe_import_context() as import_ctx:
+      from torch.optim import ...
+
+   TorchSolver = import_ctx.import_from('torch_solver', 'TorchSolver')
+
+   class Solver(TorchSolver):
+      ...
```

Comment:
I think this defeats a bit the purpose of "easily extendable" benchmark.
Maybe we should come up with a self contain example, where we declare everything in this class, without relying on the sub class? (just instanciate the optimizer, the model and call the while loop?)

Then we can explain a bit how to make it with data aug and everything using the more complex `TorchSolver`?

Reply:
Ah I see, yes why not

Reply:
actually it makes it look much more complicated I would say

we can discuss this offline maybe